spec.bs

```diff
@@ -2141,11 +2141,9 @@ of the following global objects:
       1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
         |result|'s \[[Value]] to a {{GenerateBidOutput}}.
       1. If no exception was [=exception/thrown=] in the previous step:
-        1. Let |groupHasAdComponents| be true.
-        1. If |ig|'s [=interest group/ad components=] is null, set |groupHasAdComponents| be false.
-        1. Let |possibleGeneratedBid| be the result of [=converting GenerateBidOutput to generated bid=]
-          with |generatedBidIDL|, |ig|, |expectedCurrency|, |isComponentAuction|, and |groupHasAdComponents|.
-        1. If |possibleGeneratedBid| is not failure, set |generatedBid| to it.
+        1. Set |generatedBid| be the result of [=converting GenerateBidOutput to generated bid=] with |generatedBidIDL|, |ig|, |expectedCurrency|, |isComponentAuction|, and |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/group has ad components=].
+      1. Otherwise, set |generatedBid| to failure.
+    1. If |generatedBid| is a [=generated bid=] and |generateBid|'s [=generated bid/bid=]'s [=bid with currency/value=] &le; 0, set |generateBid| to failure.
     1. If |generatedBid| is null, set it to failure.

```

Comment:
it seems to be my mistake (at least cannot remember or find why I checked null here). Now I think |generatedBid| cannot be null, since I don't see a place possibly setting it to null.

Reply:
right I see it now. Thanks